### PR TITLE
GEODE-5803: Set test facets to inherit 'test' configuration

### PR DESCRIPTION
**Pull request by PurelyApplied**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [n/a] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by PurelyApplied** [comment]
This PR is based off #2547, since that PR also addresses the removal of some inherited test configuration.

File: gradle/test.gradle [hunk 1]
```diff
@@ -35,88 +35,91 @@ gradle.taskGraph.whenReady({ graph ->
...
+    // We declare each test type to have parent source "test" so that test all test configurations
+    // (gradle parallelization thread count, core classpath, etc) are inherited.
+    
+    // TODO: During review, please comment on whether, for instance, we might consider
+    // performanceTest to be a child of distributedTest, or if a flat structure is preferable.
+    // Otherwise, each will implicitly inherit "main" 
     integrationTest {
       testTaskName = 'integrationTest'
       includeInCheckLifecycle = false
+      parentSourceSet = 'test'
```
Comment (pivotal-jbarrett) [inline gradle/test.gradle]: NO!!!! REVERT!

This causes all sorts of bad in IJ and Eclipse and falls outside the standard model for test sources. XTest sources should not depend on the outputs of other xTest sources.

Comment (PurelyApplied) [reply gradle/test.gradle]: Ahh.  Good to know.  I was thrown off by our apparent desire to have that structure as explicitly called out in `geode-core`'s build.

So would you say that this block should be considered and anti-pattern, and this ticket should be updated to address this issue instead?

```
// geode-core/build.gradle:217-223
dependencies {
  // Integration Tests
  integrationTestCompile sourceSets.test.output

  // Distributed Tests
  distributedTestCompile sourceSets.integrationTest.output
}
```

Comment (pivotal-jbarrett) [reply gradle/test.gradle]: Yes this is a mistake and should have been cleaned up months ago when we cleaned up all the other sub projects. Whoops!

**Comment by PurelyApplied** [comment]
Looking at disentangling the `geode-core` source sets as mentioned above, this appears to be significantly non-trivial (and perhaps why it wasn't done during the previous test restructure).  Closing the PR until more work can be done on that front.

**Comment by pivotal-jbarrett** [comment]
It should be pretty trivial. Having been part of most of the initial refactoring I can tell you this was simply an omission.

**Pull request closed without merging**